share/completions/rsync.fish

```diff
@@ -16,30 +32,41 @@ complete -c rsync -l suffix -d "Backup suffix (default ~ w/o --backup-dir)"
 complete -c rsync -s u -l update -d "Skip files that are newer on the receiver"
 complete -c rsync -l inplace -d "Update destination files in-place"
 complete -c rsync -l append -d "Append data onto shorter files"
+complete -c rsync -l append-verify -d "--append w/old data in file checksum"
```

Comment:
What does this mean?

I'm assuming "w/old" means "with old", but what's old data in the file checksum?

The full explanation in the man page goes into more detail:

>This special copy mode works like --append except that all the data in the file is included in the checksum verification (making it much less efficient but  also potentially  safer)

So it's "append with checksum computed over all the old data".

Unfortunately it then goes on to explain

>This option can be dangerous if you aren't 100% sure that all the files in the transfer are shared, growing files.

And now I'm confused and unsure again - is this safer or dangerous? So we check `--append`, and find out that *appending* is dangerous if not all files grow, and append-verify is a bit safer than that in confirming the checksum, but still bad if things aren't growing.

So something like "append with full file checksum" seems to communicate the basic idea.

Reply:
As far as I understand it the old data is the (smaller) file that already exists on the destination. With normal append it calculates the checksum only over the appended data that gets transferred (possibly over unstable networks).

> appending is dangerous if not all files grow, and append-verify is a bit safer than that in confirming the checksum, but still bad if things aren't growing.

I think that is correct, append is dangerous in general, but the full checksum makes it at least a bit safer.